File: n-th-tribonacci-number/n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        def helper(n, hashmap = {}):
            if n in hashmap: return hashmap[n]
            elif n == 2: return 1
            elif n < 2: return n
            else:
                hashmap[n] = helper(n - 1, hashmap) + helper(n - 2, hashmap) + helper(n - 3, hashmap)
                return hashmap[n]
        
        return helper(n)

File: n-th-tribonacci-number/n-th-tribonacci-number.js
var tribonacci = function(n) {
    const helper = (n, hashmap = {}) => {
        if (n in hashmap) {
            return hashmap[n]
        }
        else if (n == 2) {
            return 1
        }
        else if (n < 2) {
            return n
        }
        else {
            hashmap[n] = helper(n - 1) + helper(n - 2, hashmap) + helper(n - 3, hashmap)
            return hashmap[n]
        }
    };
    
    return helper(n)
};